### PR TITLE
Improve documentation for fiber-scoped variables

### PR DESCRIPTION
**Pull request by eregon**
* Especially around Enumerator.

### CODE REVIEW

**Review by ioquatix — APPROVED** [review]
Awesome, looks great.

File: enumerator.c [hunk 1]
```diff
@@ -83,15 +85,44 @@
...
+ * These methods do not affect other internal enumeration methods,
+ * unless the underlying iteration method itself has side-effect, e.g. IO#each_line.
+ *
+ * External iteration differs *significantly* from internal iteration
+ * due to using a Fiber:
+ *  - The Fiber adds some overhead compared to internal enumeration.
+ *  - The stacktrace will only include the stack from the Enumerator, not above.
+ *  - Fiber-local variables are *not* inherited inside the Enumerator Fiber,
+ *    which instead starts with no Fiber-local variables.
+ *  - Fiber-scoped variables *are* inherited and are designed
```
Comment (eregon) [inline enumerator.c:100]: NEWS talks about it like `inheritable fiber storage` but that seems difficult to use here, and also in `Fiber[]/[]=` docs. I'm not sure how to resolve this different terms for the same thing.

Comment (eregon) [reply enumerator.c:100]: Maybe `Introduce Fiber.[] and Fiber.[]= for inheritable fiber storage, also referred as Fiber-scoped variables.` in NEWS?

Comment (ioquatix) [reply enumerator.c]: Replied on Slack.

**Pull request merged**